lib/bulk-apply-applications.js

```diff
@@ -243,91 +261,166 @@ function bulkApplyApplications (args, callback) {
-                emailParents(cb);
+            //If this applicant has no email of their own
+            if (!profile.email) {
+              //move to next applicant if there are no parents for this applicant either (We have no one to email)
+              if (_.isEmpty(profile.parents)) {
+                return profileCb();
               } else {
-                return cb();
+                //we email the parent email if no email is found for this applicant
+                payload.to = profile.parents[0].email || null;
               }
-            });
-
-            function emailParents (cb) {
-              if (_.isEmpty(profile.parents)) return cb();
-              var parentsEmailed = [];
-              async.eachSeries(profile.parents, function (parent, cb) {
-                if (!_.isObject(parent)) {
-                  seneca.act({role: 'cd-users', cmd: 'load', id: parent, user: requestingUser}, function (err, parentUser) {
-                    if (err) return cb(err);
-                    payload.to = parentUser.email;
-                    if (!_.contains(parentsEmailed, payload.to) && payload.to !== profile.email) {
-                      parentsEmailed.push(payload.to);
-                      seneca.act({role: 'cd-dojos', cmd: 'send_email', payload: payload}, cb);
-                    } else {
-                      return cb();
-                    }
-                  });
+            }
+            //send the email
+            seneca.act({role: 'kue-queue', cmd: 'enqueue', name: 'bulk-apply-applications-kue', msg: _.clone({role: 'cd-dojos', cmd: 'send_email', payload: payload}, function(err, res) {
+              if (err) {
+                return profileCb(err);
+              } else {
+                return profileCb(res);
+              }
+            })});
+          });
+        },
+        //Send notification email to the dojo
+        function sendDojoEmail () {
+          //if email notifications are enabled for the event, send an email to the dojo
+          if (eventData.notifyOnApplicant) {
+            //subject for email to send to the dojo
+            emailSubject = dojoEmailSubjectData[applications[0].status];
+            //email payload changes for dojo
+            var payloadChanges = {
+              to: dojo.email || null,
+              code: dojoEmailCode,
+              subject: emailSubject,
+              from: 'The CoderDojo Team <info@coderdojo.org>',
+              content: {
+                dojoName: dojo.name
+              }
+            };
+            //create the payload for the dojo
+            var dojoPayload = _.merge(payloadChanges, commonPayload);
+            //send the dojo email
+            seneca.act({role: 'kue-queue', cmd: 'enqueue', name: 'bulk-apply-applications-kue', msg: _.clone({role: 'cd-dojos', cmd: 'send_email', payload: dojoPayload}, function (err, res) {
+                if (err) {
```

Comment:
either pass the error to done() or use the logger to indicate that something went wrong. Returning will do nothing here